libraries/plugins/xfpga/usrclk/fpga_user_clk.c

```diff
@@ -0,0 +1,825 @@
+#define  AGILEX_USRCLK_REV      1
+
+/*
+ * USER CLK CSR register definitions
+ */
+ /* Field definitions for both USERCLK_FREQ_CMD0 and USERCLK_FREQ_STS0 */
+#define IOPLL_FREQ_CMD0		    0x8
+#define IOPLL_DATA		        GENMASK_ULL(31, 0)
+#define IOPLL_ADDR		        GENMASK_ULL(41, 32)
+#define IOPLL_WRITE		        BIT_ULL(44)
+#define IOPLL_SEQ		        GENMASK_ULL(49, 48)
+#define IOPLL_AVMM_RESET_N	    BIT_ULL(52)
+#define IOPLL_MGMT_RESET	    BIT_ULL(56)
+#define IOPLL_RESET		        BIT_ULL(57)
+
+#define IOPLL_FREQ_CMD1		    0x10
+/* Field definitions for both USERCLKL_FREQ_CMD1 and USERCLK_FREQ_STS1 */
+#define IOPLL_CLK_MEASURE	    BIT_ULL(32)	/* Measure clk: 0=1x, 1=2x */
+#define IOPLL_FREQ_STS0		    0x18
+#define IOPLL_LOCKED		    BIT_ULL(60)
+#define IOPLL_AVMM_ERROR	    BIT_ULL(63)
+
+#define IOPLL_FREQ_STS1		    0x20
+#define IOPLL_FREQUENCY	        GENMASK_ULL(16, 0)	/* 10 kHz units */
+#define IOPLL_REF_FREQ	        GENMASK_ULL(50, 33)	/* Reference Freq */
+#define IOPLL_VERSION		    GENMASK_ULL(63, 60)	/* User clock version */
+
+/*
+ * Control and status registers for the IOPLL
+ * https://www.altera.com/en_US/pdfs/literature/hb/stratix-10/ug-s10-clkpll.pdf
+ * Section 7.2
+ */
+
+#define CFG_PLL_LOW			    GENMASK_ULL(7, 0)
+#define CFG_PLL_HIGH			GENMASK_ULL(15, 8)
+#define CFG_PLL_BYPASS_EN		BIT_ULL(16)
+#define CFG_PLL_EVEN_DUTY_EN    BIT_ULL(17)
+
+#define PLL_EVEN_DUTY_EN_SHIFT	7
+
+#define PLL_N_HIGH_ADDR			0x100
+#define PLL_N_BYPASS_EN_ADDR    0x101 /* Same as PLL_CP1_ADDR */
+#define PLL_N_EVEN_DUTY_EN_ADDR	0x101 /* Same as PLL_CP1_ADDR */
+#define PLL_N_LOW_ADDR			0x102
+
+#define PLL_M_HIGH_ADDR			0x104
+#define PLL_M_BYPASS_EN_ADDR	0x105
+#define PLL_M_EVEN_DUTY_EN_ADDR	0x106
+#define PLL_M_LOW_ADDR			0x107
+
+#define PLL_C0_HIGH_ADDR		0x11b
+#define PLL_C0_BYPASS_EN_ADDR	0x11c
+#define PLL_C0_EVEN_DUTY_EN_ADDR 0x11d
+#define PLL_C0_LOW_ADDR			0x11e
+
+#define PLL_C1_HIGH_ADDR		0x11f
+#define PLL_C1_BYPASS_EN_ADDR	0x120
+#define PLL_C1_EVEN_DUTY_EN_ADDR 0x121
+#define PLL_C1_LOW_ADDR			0x122
+
+#define CFG_PLL_CP1			    GENMASK_ULL(2, 0)
+#define PLL_CP1_ADDR			0x101 /* Same as PLL_N_BYPASS_EN_ADDR */
+#define PLL_CP1_SHIFT			4
+
+#define CFG_PLL_LF			    GENMASK_ULL(13, 6)
+#define PLL_LF_ADDR			    0x10a
+#define PLL_LF_SHIFT			3
+
+#define CFG_PLL_CP2			    GENMASK_ULL(5, 3)
+#define PLL_CP2_ADDR			0x10d
+#define PLL_CP2_SHIFT			5
+
+#define CFG_PLL_RC			    GENMASK_ULL(1, 0)
+#define PLL_RC_SHIFT			1
+
+#define PLL_REQUEST_CAL_ADDR	0x149
+#define PLL_REQUEST_CALIBRATION	BIT(6)
+
+#define PLL_ENABLE_CAL_ADDR		0x14a
+#define PLL_ENABLE_CALIBRATION	0x03
+
+#define IOPLL_MEASURE_LOW		0
+#define IOPLL_MEASURE_HIGH		1
+#define IOPLL_MEASURE_DELAY_MS	4000
+#define IOPLL_RESET_DELAY_MS	1000
+#define IOPLL_CAL_DELAY_MS		1000
+
+#define IOPLL_WRITE_POLL_INVL_US	10	/* Write poll interval */
+#define IOPLL_WRITE_POLL_TIMEOUT_US	1000000	/* Write poll timeout */
+
+
+
+fpga_result usrclk_reset(uint8_t* uio_ptr)
```

Comment:
We should turn these new files on in test-codingstyle-all.sh and fix any issues.

Reply:
ran test-codingstyle-all.sh c and fixed all coding style issues